socs/agents/wiregrid_tiltsensor/drivers/dwl.py

```diff
@@ -0,0 +1,146 @@
+# Built-in python modules
+import time as tm
+
+# Control modules
+from socs.common import moxa_serial as mx
+
+
+class DWL:
+    """
+    The DWL object is for communicating with the DWL-5000XY gravity sensor
+
+    Args:
+    tcp_ip (str): TCP IP address
+    tcp_port (int): TCP port
+    """
+    waittime = 0.05  # sec
+
+    def __init__(self, tcp_ip=None, tcp_port=None, timeout=None, isSingle=False, verbose=0):
+        self.tcp_ip = tcp_ip
+        self.tcp_port = tcp_port
+        self.isSingle = isSingle
+        self.verbose = verbose
+
+        # Connect to device
+        msg = self.__conn(tcp_ip, tcp_port, timeout)
+        print(msg)
+
+    def __del__(self):
+        print(f"Disconnecting from TCP IP {self.tcp_ip} at port {self.tcp_port}")
+        self.ser.close()
+        return
+
+    def get_angle(self):
+        """ Measure the single-axis or two-axis angle """
+        self.clean_serial()
+        if self.isSingle:
+            command = b"\x06\x01\x01\xAA\x00\x00\x00\x00\x00\x00\x00\x00"
+        else:
+            command = b"\x06\x01\x02\xAA\x00\x00\x00\x00\x00\x00\x00\x00"
+        if self.verbose > 0:
+            print(f'get_angle() command = {command}')
+
+        read = []
+        SIZE = 12
+
+        # write and read serial
+        self.ser.write(command)
+        read_hex = self.ser.read(SIZE)
+        if self.verbose > 0:
+            print(f'{read_hex}')
+        read = [hex(r) for r in read_hex]
+        if self.verbose > 0:
+            print(f'read_hex = {read_hex}')
+            print(f'read = {read}')
+
+        # check header matching and calculate the angles
+        if self.isSingle:
+            header = ['0x61', '0x11']
+        else:
+            header = ['0x61', '0x22']
+        if read[0:2] == header:
+            readInt = []
+            val = ()
+            for c in read:
+                readInt.append((int)(c, 16))
```

Comment:
Does `self.ser.read(SIZE)` always return something? It seems like it would be possible to get nothing back from the read and end up with an empty `readInt`, which would cause issues down on lines 67, 78, 89, etc. I'd suggest adding some check that `read` is the expected size.

Reply:
Your concern is very reasonable. I modified it to return -999 or (-999, -999) if `read` does not have the expected size. (022b0cc)